qualtran/resource_counting/_call_graph.py

```diff
@@ -76,19 +76,19 @@ def _generalize_callees(
     This calls `generalizer` on each of the callees returned from that function,
     and filters out cases where `generalizer` returns `None`.
     """
-    callee_counts: List[BloqCountT] = []
+    callee_counts: Dict[Bloq, Union[int, sympy.Expr]] = defaultdict(lambda: 0)
     for callee, n in raw_callee_counts:
         generalized_callee = generalizer(callee)
         if generalized_callee is None:
             # Signifies that this callee should be ignored.
             continue
-        callee_counts.append((generalized_callee, n))
-    return callee_counts
+        callee_counts[generalized_callee] += n
+    return list(callee_counts.items())
 
 
 def get_bloq_callee_counts(
     bloq: 'Bloq',
-    generalizer: Optional['GeneralizerT'] = None,
+    generalizer: Optional[Union['GeneralizerT', Sequence['GeneralizerT']]] = None,
```

Comment:
Should we take the opportunity to change `get_bloq_callee_counts` and ` _generalize_callees` to a `Mapping` / `Dict` as well? The conversion from Dict -> List -> Dict seems unnecessary.

Reply:
let's do that all in the same pr